build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1407,6 +1412,11 @@ message ExecuteOperationMetadata {
   // being executed.
   Digest action_digest = 2;
 
+  // A unique identifier for an ongoing execution of the action, which will
+  // also be returned as `ActionResult.execution_metadata.execution_uuid`.
+  // If the execution is restarted, the server SHOULD provide a new identifier.
+  string execution_uuid = 5;
```

Comment:
Maybe it's not important, but still worth asking: at what point in time do we generate an `execution_uuid`? Does a `CACHE_CHECK` or `QUEUED` operation already have a non-empty `execution_uuid`, or does it only apply to operations in stages `EXECUTING` or `COMPLETED`?

Reply:
I thought about it but realized it might be up to the server side to decide. We would mention that in the description.

Reply:
I'd be inclined towards a semantic of generating an ID when the action starts to execute and not before.

Reply:
I would like to leave it open for the server to decide, but put an expectation on it being provided during `EXECUTING` and `COMPLETED`. Should this field be mandatory in v3 but marked as optional using "if set, ..." for v2?